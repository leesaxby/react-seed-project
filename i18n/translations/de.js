export default {
    'todo.filter.todo': 'alles',
    'todo.filter.showDone': 'Show fertig',
    'todo.filter.ariaLabel': 'filter auswählen',
    'todo.form.addItem': 'Artikel hinzufügen',
};
