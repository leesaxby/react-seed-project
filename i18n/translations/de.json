{
  "todo.filter.ariaLabel": "filter auswählen",
  "todo.filter.showDone": "Show fertig",
  "todo.filter.todo": "alles",
  "todo.form.addItem": "Artikel hinzufügen"
}