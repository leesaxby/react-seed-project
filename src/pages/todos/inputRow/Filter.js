import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, injectIntl, defineMessages } from 'react-intl';
import { ToggleButtonGroup, ToggleButton } from 'react-bootstrap';

const messages = defineMessages({
    'todo.filter.ariaLabel': {
        id: 'todo.filter.ariaLabel',
        description: 'Label for the todo filter choice',
        defaultMessage: 'choose filter'
    },
    'todo.filter.todo': {
        id: 'todo.filter.todo',
        description: 'Label for the button that filters active todos',
        defaultMessage: 'todo'
    },
    'todo.filter.done': {
        id: 'todo.filter.done',
        description: 'Label for the button that filters complete todos',
        defaultMessage: 'done'
    }
});

export class Filter extends React.Component {

    static propTypes = {
        filter: PropTypes.oneOf([ 'ACTIVE', 'DONE' ]).isRequired,
        onUpdateFilter: PropTypes.func.isRequired,
        intl: PropTypes.shape({
          formatMessage: PropTypes.func.isRequired
        }).isRequired
    };

    render() {
        const { filter, onUpdateFilter, intl } = this.props;
        return(
            <ToggleButtonGroup type="radio"
                               name="filterToggle"
                               role="radiogroup"
                               aria-label={intl.formatMessage(messages['todo.filter.ariaLabel'])}
                               value={filter}
                               onChange={onUpdateFilter}>

                <ToggleButton value="ACTIVE"
                              role="radio"
                              aria-checked={String(filter === 'ACTIVE')}>
                    <FormattedMessage {...messages['todo.filter.todo']} />
                </ToggleButton>

                <ToggleButton value="DONE"
                              role="radio"
                              aria-checked={String(filter === 'DONE')}>
                    <FormattedMessage {...messages['todo.filter.done']} />
                </ToggleButton>

            </ToggleButtonGroup>
        );
    }
}

export default injectIntl(Filter);
