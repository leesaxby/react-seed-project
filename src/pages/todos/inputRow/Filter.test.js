import React from 'react';
import { shallow } from 'enzyme';
import createComponentWithIntl from 'Test/createComponentWithIntl';
// TODO: Add webpack alias for translations.
import de from '../../../../i18n/translations/de';
import { Filter } from './Filter';

describe('A Filter', () => {

	function setup() {
		const props = {
		    onUpdateFilter: jest.fn(),
		    intl: {
		      formatMessage: jest.fn(),
		    },
		};
		const wrapper = shallow(<Filter { ...props } />);

		return {
			props,
			wrapper,
		};
	}

	it('renders component', () => {
		const { props } = setup();
		const tree = createComponentWithIntl(<Filter { ...props } />).toJSON();

		expect(tree).toMatchSnapshot();
	  });


    it('selects a button based on the value of its \'filter\' prop', () => {
		const { props } = setup();
		const updatedProps = Object.assign({}, props, { filter: 'DONE' });
		const tree = createComponentWithIntl(<Filter { ...updatedProps } />).toJSON();

		expect(tree).toMatchSnapshot();
    });

    it('triggers a callback when the selected button is changed', () => {
    	const { wrapper } = setup();

    	expect(wrapper.instance().props.onUpdateFilter).not.toHaveBeenCalled();

    	wrapper.simulate('change', 'DONE');

    	expect(wrapper.instance().props.onUpdateFilter).toHaveBeenCalledWith('DONE');
    });

    it('displays correct translations', () => {
		const { props } = setup();
		const tree = createComponentWithIntl(
			<Filter { ...props } />,
			{
				locale: 'de',
				messages: de,
			}
		).toJSON();

		expect(tree).toMatchSnapshot();
    });


});
