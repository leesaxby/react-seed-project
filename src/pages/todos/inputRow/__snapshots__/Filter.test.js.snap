// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Filter displays correct translations 1`] = `
<div>
  <span>
    Show fertig
  </span>
  <label
    htmlFor="todoStateSwitch"
  >
    <span
      className="MuiSwitch-root-1"
    >
      <span
        aria-label={undefined}
        className="MuiButtonBase-root-21 MuiIconButton-root-15 MuiSwitchBase-root-10 MuiSwitchBase-default-12 MuiSwitch-default-5"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={undefined}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label-20"
        >
          <span
            className="MuiSwitch-icon-3"
          />
          <input
            checked={false}
            className="MuiSwitchBase-input-11"
            disabled={undefined}
            id="todoStateSwitch"
            name={undefined}
            onChange={[Function]}
            tabIndex={undefined}
            type="checkbox"
            value={undefined}
          />
        </span>
        <span
          className="MuiTouchRipple-root-23"
        />
      </span>
      <span
        className="MuiSwitch-bar-2"
      />
    </span>
  </label>
</div>
`;

exports[`A Filter renders component 1`] = `
<div>
  <span>
    Show Done
  </span>
  <label
    htmlFor="todoStateSwitch"
  >
    <span
      className="MuiSwitch-root-1"
    >
      <span
        aria-label={undefined}
        className="MuiButtonBase-root-21 MuiIconButton-root-15 MuiSwitchBase-root-10 MuiSwitchBase-default-12 MuiSwitch-default-5"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={undefined}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label-20"
        >
          <span
            className="MuiSwitch-icon-3"
          />
          <input
            checked={false}
            className="MuiSwitchBase-input-11"
            disabled={undefined}
            id="todoStateSwitch"
            name={undefined}
            onChange={[Function]}
            tabIndex={undefined}
            type="checkbox"
            value={undefined}
          />
        </span>
        <span
          className="MuiTouchRipple-root-23"
        />
      </span>
      <span
        className="MuiSwitch-bar-2"
      />
    </span>
  </label>
</div>
`;

exports[`A Filter selects a button based on the value of its 'filter' prop 1`] = `
<div>
  <span>
    Show Done
  </span>
  <label
    htmlFor="todoStateSwitch"
  >
    <span
      className="MuiSwitch-root-1"
    >
      <span
        aria-label={undefined}
        className="MuiButtonBase-root-21 MuiIconButton-root-15 MuiSwitchBase-root-10 MuiSwitchBase-default-12 MuiSwitch-default-5 MuiSwitchBase-checked-13 MuiSwitch-checked-6 MuiSwitch-checkedPrimary-7"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={undefined}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label-20"
        >
          <span
            className="MuiSwitch-icon-3 MuiSwitch-iconChecked-4"
          />
          <input
            checked={true}
            className="MuiSwitchBase-input-11"
            disabled={undefined}
            id="todoStateSwitch"
            name={undefined}
            onChange={[Function]}
            tabIndex={undefined}
            type="checkbox"
            value={undefined}
          />
        </span>
        <span
          className="MuiTouchRipple-root-23"
        />
      </span>
      <span
        className="MuiSwitch-bar-2"
      />
    </span>
  </label>
</div>
`;
